src/client/connect/http.rs

```diff
@@ -228,6 +230,25 @@ impl<R> HttpConnector<R> {
         self
     }
 
+    /// Sets the value for the `SO_BINDTODEVICE` option on this socket.
+    ///
+    /// If a socket is bound to an interface, only packets received from that particular
+    /// interface are processed by the socket. Note that this only works for some socket
+    /// types, particularly AF_INET sockets.
+    ///
+    /// On Linux it can be used to specify a [VRF], but the binary needs
+    /// to either have `CAP_NET_RAW` or to be run as root.
+    ///
+    /// This function is only available on Android、Fuchsia and Linux.
+    ///
+    /// [VRF]: https://www.kernel.org/doc/Documentation/networking/vrf.txt
+    #[cfg(any(target_os = "android", target_os = "fuchsia", target_os = "linux"))]
+    #[inline]
+    pub fn bind_device<S: ToString>(&mut self, interface: S) -> &mut Self {
```

Comment:
How about the name `set_interface`?

Reply:
Also, I'd probably say the generic should be `S: Into<String>`, since otherwise `set_interface(format("eth{}", num))` will make a string with `format!`, and then copy thanks to `ToString`.

Reply:
I agree with you